hai/parallel.py

```diff
@@ -238,3 +216,59 @@ def exceptions(self) -> Dict[str, Exception]:
             for t in self.tasks
             if t.ready() and not t._success  # type: ignore[attr-defined]
         }
+
+
+class ChordParallelRun(BaseParallelRun):
+    """Parallel run that inherits the thread pool from another parallel run
+    instead of managing its own
+    """
+
+    def __init__(self, main_run: "ParallelRun") -> None:
+        super().__init__()
+        self.main_run = main_run
+        self.pool = main_run.pool
+
+    def ready(self) -> bool:
+        return all([t.ready() for t in self.tasks])
```

Comment:
```suggestion
        return all(t.ready() for t in self.tasks)
```
to avoid calling `t.ready()` on all tasks if any one of them is not ready?